WcaOnRails/app/models/competition.rb

```diff
@@ -466,22 +466,76 @@ def country
   end
 
   def self.search(query, params: {})
-    sql_query = "%#{query}%"
-    Competition.where("id LIKE :sql_query OR name LIKE :sql_query OR cellName LIKE :sql_query OR cityName LIKE :sql_query OR countryId LIKE :sql_query", sql_query: sql_query).order(year: :desc, month: :desc, day: :desc)
+    competitions = Competition.where(showAtAll: true)
```

Comment:
I've had a fair amount of confusion about whether a query has been executed or not, and problems with the actual query being made at a different time than when the programmer expected it.

They do share part of the interface. I can't remember the concrete problems years ago, but the fact that `.each` works on both the Relation and a result was probably a factor.
----
Only a few simple expressions can be expressed "natively" in AR: http://guides.rubyonrails.org/active_record_querying.html#hash-conditions

Even something as basic as comparison requires writing raw (or at least undercooked) SQL (`MyModel.where("length > ?", params[:min_length])`). A Mongo like interface could be `MyModel.where(length: {lt:  params[:min_length]})`. In that world you could build your conditions programmatically and not have to concatenate SQL strings like a savage.

Also, AR didn't even support OR last I checked. It's like they're not even trying!

I didn't come here to complain, but you did ask...
----
Yeah, that boilerplate is a problem. I'd make it part of an ApiField set of classes. The class would know how to do `Date.safe_parse` or whatever, and also how to make the query. The rest can be a template method in the base class. Probably not worth doing right now.

Personally I'd switch to `raise <exception> unless <result>` to make the boilerplate smaller, but that's just personal preference.